clients/ts/signalr/src/ServerSentEventsTransport.ts

```diff
@@ -56,7 +56,11 @@ export class ServerSentEventsTransport implements ITransport {
                 return;
             }
 
-            const eventSource = new this.eventSourceConstructor(url, { withCredentials: true });
+            let cookies: string = "";
+            if ((this.httpClient as any).getCookies) {
+                cookies = (this.httpClient as any).getCookies(url);
+            }
+            const eventSource = new this.eventSourceConstructor(url, { withCredentials: true, headers: { Cookie: cookies } } as EventSourceInit);
```

Comment:
Can you explicitly avoid providing `headers` if we're in a browser (`typeof window !== "undefined"`)? I don't want it to weird out if browsers add some support for this but, for example, throw if the `Cookie` header is set manually (or something similar).